### ISSUE TITLE
API/Webservice to allow course completion date to be set for users

### ISSUE THREAD

**Issue by akoharek** [danmarsden/moodle-local_recompletion#39]
Dan, this is wonderful plugin. I wonder maybe there is a way to update the course completion date for all user in one step. Do you have any plan to introduce this feature?

**Comment by danmarsden** [comment]
Does the "Modify completion dates" page that is accessible under the "recompletion setttings" page in the course settings menu do what you need?

**Comment by akoharek** [comment]
hi Dan,

yes, partially. We have a task to migrate some thousands of completion date into Moodle from an other system. Using the gui is one option for few users and date combination. Sorry for being not clear in my first entry, what we are looking for is an api or csv upload possibility to make it.

**Comment by danmarsden** [comment]
Makes sense! - I'm not currently working on anything similar. If you have development capabilities, pull requests are always welcome (it just may take me some time to review/comment/merge them) - alternatively feel free to get in touch privately if you would like to fund further improvments  (I work for the Moodle Partner Catalyst IT)

**Comment by akoharek** [comment]
sent you a pm

**Comment by danmarsden** [comment]
new webservice now in place which allows user completion date to be set.
